### ISSUE TITLE
Restore caching of Pipenv environment in python-ci?

### ISSUE THREAD

**Issue by rod-glover** [pacificclimate/crmprtd#118]
Recently, the caching step of python-ci started causing errors in invoking Pipenv. The step was removed.

It may be possible to restore caching, but using a different method. The caching step included a deprecation message as follows:

```
Run actions/cache@v2
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-[1](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:6:1)0-11-github-actions-deprecating-save-state-and-set-output-commands/
Received 27703[8](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:6:9)89 of 27703889 (100.0%), 41.0 MBs/sec
Cache Size: ~26 MB (27703889 B)
/usr/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/b9dcfc55-6fc6-4d2d-aba1-8150aeec4adf/cache.tzst -P -C /home/runner/work/crmprtd/crmprtd
Cache restored successfully
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
Cache restored from key: Linux-pipenv-0766b0f[9](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:6:10)ec5793bc586a22b45cc8aac4[1](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:8:1)f3dd0d07b0[8](https://github.com/pacificclimate/crmprtd/achttps://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
tions/runs/3661714606/jobs/6190184637#step:8:9)47c0[9](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:8:10)268276699cd6[19](https://github.com/pacificclimate/crmprtd/actions/runs/3661714606/jobs/6190184637#step:8:20)f
```

It may be worth trying to restore caching using the reommended approach to saving state etc. in the [supplied link](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
).

**Comment by rod-glover** [comment]
Here's the commit where caching was removed: https://github.com/pacificclimate/crmprtd/pull/116/commits/aa183b89e169e46a01e8b5d6b62d9ccecee1062d

**Comment by jameshiebert** [comment]
Can be closed given #161
